improver/cli/__init__.py

```diff
@@ -239,27 +243,47 @@ def constrained_inputcubelist_converter(to_convert):
                 than one match is found.
         """
         from improver.utilities.load import load_cube
-        from iris.cube import CubeList
-
-        cubelist = CubeList()
-        for constr in constraints:
-            constr_list = [constr] if isinstance(constr, str) else constr
-            found_cubes = []
-            for constr_item in constr_list:
-                try:
-                    found_cubes.append(maybe_coerce_with(
-                        load_cube, to_convert, constraints=constr_item))
-                except ValueError:
-                    pass
-            if len(found_cubes) != 1:
-                msg = (f"Incorrect number of valid inputs available for the "
-                       "{constr} constraint. "
-                       f"Number of valid inputs: {len(found_cubes)} "
-                       f"The valid inputs found are: {found_cubes}")
-                raise ValueError(msg)
-            cubelist.extend(found_cubes)
-
-        return cubelist
+        from iris import load_raw
+
+        # Load cube if given a string.
+        if isinstance(to_convert, str):
+            cubelist = load_raw(to_convert)
+            is_str = True
+        else:
+            cubelist = to_convert
+            is_str = False
```

Comment:
I think I require some explanation as to why we've had to abandon ```maybe_coerce_with``` for differentiating between cubes and strings?

Reply:
`maybe coerce with` passed on a string to a given function to load, or returned an object.
That will not work if we want to load with constraints at the point as you don't want to just return the cubelist but you want to return a subcube of that list.